### PR TITLE
e2ee: stage 2

### PR DESCRIPTION
**Pull request by saghul**
Adapt to E2EE changes in lib-jitsi-meet. Notably:

---
    e2ee: introduce per-participant randomly generated keys

    This the second stage in our E2EE journey.

    Instead of using a single pre-shared passphrase for deriving the key used for
    E2EE, we now establish a secure E2EE communication channel amongst peers.

    This channel is implemented using libolm, using XMPP groupchat or JVB channels
    as the transport.

    Once the secure E2EE channel has been established each participant will generate
    a random 32 byte key and exchange it over this channel.

    Keys are rotated (well, just re-created at the moment) when a participant joins
    or leaves.
---

### CODE REVIEW

File: react/features/base/lastn/middleware.js [hunk 1]
```diff
@@ -71,11 +71,13 @@ function _updateLastN({ getState }) {
     const defaultLastN = typeof config.channelLastN === 'undefined' ? -1 : config.channelLastN;
     let lastN = defaultLastN;
 
-    // Apply last N limit based on the # of participants
-    const limitedLastN = limitLastN(participantCount, lastNLimits);
+    if (lastNLimits) {
```
Comment (fippo) [inline react/features/base/lastn/middleware.js:74]: unrelated?

Comment (saghul) [reply react/features/base/lastn/middleware.js:74]: Yeah, just a quick thing while Pawel pushes the proper fix in a PR :-)

**Review by damencho — APPROVED** [review]

**Review by initpwn — COMMENTED** [review]
Does the older commit implement ECDSA?
